corgidrp/walker.py

```diff
@@ -53,36 +53,51 @@ def walk_corgidrp(filelist, CPGS_XML_filepath, outputdir, template=None):
         filelist (list of str): list of filepaths to files
         CPGS_XML_filepath (str): path to CPGS XML file for this set of files in filelist
         outputdir (str): output directory folderpath
-        template (str or json): custom template. either the full json file, or a filename of
-                                a template that's already in the recipe_templates folder
+        template (str or json): custom template. It can be one of three things
+                                  * the full json object, 
+                                  * a filename of a template that's already in the recipe_templates folder
+                                  * a filepath to a template on disk somewhere
+                                
 
     Returns:
-        json: the JSON recipe that was used for processing
+        json or list: the JSON recipe (or list of JSON recipes) that was used for processing
     """
     if isinstance(template, str):
-        recipe_filepath = os.path.join(recipe_dir, template)
+        if os.path.sep not in template:
+            # this is just a template name in the recipe_templates folder
+            recipe_filepath = os.path.join(recipe_dir, template)
+        else:
+            recipe_filepath = template
+        
         template = json.load(open(recipe_filepath, 'r'))
 
     # generate recipe
-    recipe = autogen_recipe(filelist, outputdir, template=template)
+    recipes = autogen_recipe(filelist, outputdir, template=template)
 
-    # process_recipe
-    run_recipe(recipe)
+    # process recipe
+    if isinstance(recipes, list):
+        # if multiple recipes
+        for recipe in recipes:
+            run_recipe(recipe)
+    else:
+        # process single recipe
+        run_recipe(recipes)
 
-    return recipe
+    return recipes
```

Comment:
I think we maybe want a unit test for when there's two recipes, no?

Reply:
I added a unit test to a the autogen_recipe() function, as it wasn't straigthforward to test this function, and I didn't want to figure it out right now.

Reply:
I made a follow Issue #255